qir/cli-tests/test_profile_cli_config.py

```diff
@@ -0,0 +1,114 @@
+        "--no-allow-internal-calls",
+        "--no-allow-poison",
+        "--no-allow-undef",
+        "--no-allow-internal-calls",
+        "--use-static-result-allocation",
+        "--use-static-qubit-allocation",
+        "--use-static-qubit-array-allocation",
+    ],
+    "provider_7ee0": [
+        "--apply",
+        "--verify-module",
+        "--validate",
+        "--disable-grouping",
+        "--always-inline",
+        "--reindex-qubits",
+        "--requires-qubits",
+        "--requires-results",
+        "--no-disable-record-output-support",
+        "--unroll-loops",
+        "--reuse-qubits",
+        "--entry-point-attr",
+        "EntryPoint",
+        "--no-allow-internal-calls",
+        "--no-allow-poison",
+        "--no-allow-undef",
+        "--no-allow-internal-calls",
+        "--use-static-result-allocation",
+        "--use-static-qubit-allocation",
+        "--use-static-qubit-array-allocation",
+    ],
+    "provider_b340": [
+        "--apply",
+        "--verify-module",
+        "--validate",
+        "--disable-grouping",
+        "--always-inline",
+        "--reindex-qubits",
+        "--requires-qubits",
+        "--requires-results",
+        "--no-disable-record-output-support",
+        "--unroll-loops",
+        "--replace-qubit-on-reset",
+        "--pull-records-back",
+        "--entry-point-attr",
+        "EntryPoint",
+        "--no-allow-internal-calls",
+        "--no-allow-poison",
+        "--no-allow-undef",
+        "--no-allow-internal-calls",
+        "--use-static-result-allocation",
+        "--use-static-qubit-allocation",
+        "--use-static-qubit-array-allocation",
+    ],
+}
+
+
+QAT_BINARY = sys.argv[1]
+fail = False
+if __name__ == "__main__":
+    for profile, args in TESTS.items():
+        cmd1 = "{} {}  --profile generic --dump-config".format(
+            QAT_BINARY, " ".join(args)
+        )
+
+        p = subprocess.Popen(
+            [QAT_BINARY, "--profile", profile, "--dump-config"] + args,
+            stdout=subprocess.PIPE,
+            stdin=subprocess.PIPE,
+        )
+        out1 = p.communicate()[0].decode("utf-8")
+        out1 = out1.replace(": generic", ": {}".format(profile))
+
+        p = subprocess.Popen(
+            [QAT_BINARY, "--profile", profile, "--dump-config"],
+            stdout=subprocess.PIPE,
+            stdin=subprocess.PIPE,
+        )
+        out2 = p.communicate()[0].decode("utf-8")
+
+        if out1 == out2:
+            print("Testing " + profile + " " + "." * (20 - len(profile)) + "[  OK  ]")
+        else:
+            fail = True
+            print("Testing " + profile + " " + "." * (20 - len(profile)) + "[ FAIL ]")
+            lines1 = out1.split("\n")
+            lines2 = out2.split("\n")
+
+            for i in range(len(lines1)):
+                l1 = lines1[i]
+                l2 = lines2[i]
+
+                if l1 != l2:
+                    print("- {}".format(l1))
+                    print("+ {}".format(l2))
```

Comment:
Consider renaming the identifiers. In some editors/fonts the identifiers `l2` looks like the number `12`.

Reply:
Updated.